libs/SmartStore/SmartStore/Classes/SFSmartStoreDatabaseManager.m

```diff
@@ -199,15 +199,14 @@ + (FMDatabase*) setKeyForDb:(FMDatabase*) db key:(NSString *)key salt:(NSString
 
 + (FMDatabase*) unlockDatabase:(FMDatabase*)db key:(NSString*)key salt:(NSString *)salt {
     if ([db open]) {
-        // Using sqlcipher 3.x default settings
-        // => should open 3.x databases without any migration
-        [[db executeQuery:@"PRAGMA cipher_default_compatibility = 3"] close];
         // Using sqlcipher 2.x kdf iter because 3.x default (64000) and 4.x default (256000) are too slow
-        // => should open 2.x databases without any migration
         [[db executeQuery:@"PRAGMA cipher_default_kdf_iter = 4000"] close];
        
         if (key)
            [db setKey:key];
+
+        // Migrating and upgrading an existing database in place (preserving data and schema) if necessary
+        [[db executeQuery:@"PRAGMA cipher_migrate"] close];
```

Comment:
@wmathurin Has this to be run everytime a DB is opened?

Reply:
To avoid running it for newly created database, I would have to change the signature of a bunch of methods to indicate the db was just created.
To avoid running for an existing database that has the right settings, I would have to check the settings first etc.

Actually looking at SQLCipher sources, it does nothing if it is already the right version. See [here](https://github.com/sqlcipher/sqlcipher/blob/v4.2.0/src/crypto_impl.c#L1353).